src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -98,40 +106,123 @@ public bool TryUnparse(StringBuilder sb)
             }
         }
 
+        public sealed class ColumnInfo
```

Comment:
>ColumnInfo [](start = 28, length = 10)

Can you refactor it slightly?
First part of this class is great, but then you start introduce internal properties, and I don't like them.
This should contain only parameters for transformation. If you need class which tied together this parameters and current schema, I would suggest you to come up with new class or maybe use TransformInfo?

Reply:
+1. Be aware that the same transformer will be applied to different data, so you cannot really attach a 'schema' to the data.
The pattern we have been following was:
- Create `ColumnInfo` inside `Transformer` to hold all the data that is serialized (like names, seeds etc. etc.)
- Create `ColInfo` inside `Mapper` to hold all the parameters that are tied to schema (column indices, actual types, metadata, etc.)

---
In reply to: [227055082](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227055082) [](ancestors = 227055082)

Reply:
Moved schema related information out of ColumnInfo and into Mapper.ColumnSchemaInfo.

Functionality-wise, this is the same as before. It's more of a cosmetic change.

---
In reply to: [227171946](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227171946) [](ancestors = 227171946,227068248,227055082)